tests/utils.sh

```diff
@@ -263,6 +263,25 @@ function create_invalid_file_path()
   echo "$invalid_path"
 }
 
+function compare_array_values()
+{
+  #shellcheck disable=SC2178
+  local -n expected="$1"
+  local -n result_to_compare="$2"
+  local line="$3"
+
+  line=${line:-0}
+
+  equal=$(echo "${expected[@]}" "${result_to_compare[@]}" | tr ' ' '\n' | sort | uniq -u)
+  if [[ -n "$equal" ]]; then
+    assertNull "$line: Arrays are not equal" "$equal"
+    echo "Diff:"
+    echo "-----"
+    echo "$equal"
+    echo "-----"
```

Comment:
I've slowly started to change change all `echo`s to `printf`s, consider changing them here.
Also, use single quotes where appropriate.